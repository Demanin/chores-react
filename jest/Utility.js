import crypto from 'crypto';

const getRandomString = (length = 20) => {
  return crypto.randomBytes(length).toString('hex');
};

const getRandomNumber = (min = 0, max = 9999999) => {
  min = Math.ceil(min);
  max = Math.floor(max);

  return Math.floor(Math.random() * (max - min)) + min;
};

const getRandomBool = () => {
  return !!getRandomNumber(0, 1);
};

export default {
  getRandomString,
  getRandomNumber,
  getRandomBool,
};
