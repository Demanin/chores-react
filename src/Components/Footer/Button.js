import React, { Component } from 'react';
import { Button as BsButton } from 'react-bootstrap';
import PropTypes from 'prop-types';

class Button extends Component
{
  render() {
    return (
      <BsButton
        disabled={this.props.active}
        onClick={() => this.props.onClick()}
      >
        {this.props.children}
      </BsButton>
    );
  }
}

Button.propTypes = {
  active: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.string,
};

export default Button;
