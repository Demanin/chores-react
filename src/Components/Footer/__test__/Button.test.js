import Button from 'src/Components/Footer/Button';
import React from 'react';
import Utility from 'jest/Utility';
import { shallow } from 'enzyme';

describe('Button', () => {
  let onClick;

  beforeEach(() => {
    onClick = jest.fn();
  });

  test('empty button snapshot', () => {
    const fixture = shallow(<Button {...{ onClick }}></Button>);

    expect(fixture).toMatchSnapshot();
  });

  test('button is disabled if active', () => {
    const active = Utility.getRandomBool();

    const fixture = shallow(<Button {...{ onClick, active }}></Button>);

    expect(fixture.prop('disabled')).toEqual(active);
  });

  test('click calls onClick', () => {
    const fixture = shallow(<Button {...{ onClick }}></Button>);

    fixture.simulate('click');

    expect(onClick).toHaveBeenCalled();
  });

  test('children applied as fixture text', () => {
    const children = Utility.getRandomString();

    const fixture = shallow(<Button {...{ onClick }}>{children}</Button>);

    expect(fixture.text()).toEqual(children);
  });
});
