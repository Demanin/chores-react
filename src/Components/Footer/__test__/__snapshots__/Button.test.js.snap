// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button empty button snapshot 1`] = `ShallowWrapper {}`;
